worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/FileWriter.java

```diff
@@ -372,7 +373,9 @@ protected void takeBuffer() {
     }
 
     // real action
-    flushBuffer = flusher.takeBuffer();
+    synchronized (flushLock) {
```

Comment:
Caller site of `takeBuffer()` is already inside `flushLock`'s synchronization block, so I think previously there is no concurrent bug here.